src/full/Agda/TypeChecking/Serialise/Instances/Common.hs

```diff
@@ -214,13 +215,12 @@ instance EmbPrj TopLevelModuleName where
   value = valueN TopLevelModuleName
 
 instance {-# OVERLAPPABLE #-} EmbPrj a => EmbPrj [a] where
-  icod_ xs = icodeNode =<< mapM icode xs
-  value    = vcase (mapM value)
---   icode []       = icode0'
---   icode (x : xs) = icode2' x xs
---   value = vcase valu where valu []      = valu0 []
---                            valu [x, xs] = valu2 (:) x xs
---                            valu _       = malformed
+  icod_ xs = icodeNode =<< go xs where
+    go :: [a] -> S Node
+    go []     = pure Empty
+    go (a:as) = do {n <- icode a; ns <- go as; pure $! Cons n ns}
```

Comment:
Compare this with the stye above where `n` and `ns` are bound strictly but we would have `pure $ Cons n ns` (no strict application).
Are both equivalent?

Reply:
They are, because `Cons` has strict fields. Using `Strict` in these modules would be IMO fine, instead of getting bangs right (and there are many equivalent ways of doing so).